### PR TITLE
WMS cleanup and guards

### PR DESCRIPTION
**Pull request by OlegRayskiy**

### CODE REVIEW

File: src/wms/WmsUtils.ts [hunk 1]
```diff
@@ -135,7 +171,17 @@ export abstract class WmsUtils {
    * @internal
    */
   private static parseWmsLegend(legend: any): AlloyWmsCapabilitiesLayerStyleLegend | null {
-    if (legend.Format && legend.Format.startsWith('image/')) {
+    if (legend.Format && typeof legend.Format === 'string' && legend.Format.startsWith('image/')) {
```
Comment (WORMSS) [inline src/wms/WmsUtils.ts]: Is there anything we can do as a typeguard for this? `legend: any`
Also, need to test for null/undefined

File: src/wms/WmsUtils.ts [hunk 2]
```diff
@@ -118,6 +146,14 @@ export abstract class WmsUtils {
    * @internal
    */
   private static parseWmsStyle(style: any): AlloyWmsCapabilitiesLayerStyle {
+    if (
```
Comment (WORMSS) [inline src/wms/WmsUtils.ts]: Need to test for null/undfined. 
Is there any typeguards we can use for this?

File: src/wms/WmsUtils.ts [hunk 3]
```diff
@@ -74,18 +85,26 @@ export abstract class WmsUtils {
   private static parseWmsLayer(layer: any): AlloyWmsCapabilitiesLayer {
     // Find bounding box of the layer or set maximum
```
Comment (WORMSS) [inline src/wms/WmsUtils.ts]: Need to test for undefined/null

File: src/polyfills/PolyfillWms.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import WMSCapabilities from 'ol/format/WMSCapabilities.js';
```
Comment (cmcnicholas) [inline src/polyfills/PolyfillWms.ts]: can we rename this alias to `OLWMSCapabilities` it's so close to `WmsCapabilities` I almost missed it

File: src/wms/WmsUtils.ts [hunk 5]
```diff
@@ -60,25 +50,64 @@ export abstract class WmsUtils {
           .map((l) => (l.styleName === 'default' ? '' : l.styleName || ''))
           .join(','),
         TRANSPARENT: transparent,
-        BGCOLOR: options.colour ? options.colour.replace('#', '0x') : undefined,
+        BGCOLOR:
+          options.colour && ColourUtils.isValidFullHex(options.colour)
```
Comment (cmcnicholas) [inline src/wms/WmsUtils.ts]: does wms layers only support hex?

Comment (OlegRayskiy) [reply src/wms/WmsUtils.ts]: request BGCOLOUR only supports `0xRRGGBB` format afaik

Comment (cmcnicholas) [reply src/wms/WmsUtils.ts]: ok then do we need "is valid full hex" instead can't we just do `color(options.colour).hex()` ?

Comment (OlegRayskiy) [reply src/wms/WmsUtils.ts]: sure

File: src/polyfills/PolyfillWms.ts [hunk 6]
```diff
@@ -12,11 +11,15 @@ export abstract class PolyfillWms {
...
+    return {
+      Title: parsed.Service.Title,
+      Layer: parsed.Capability.Layer,
+    };
   }
   /**
    * Private singleton instance of Wms capabilities reader
    */
-  private static caps = new WMSCapabilities();
+  private static caps = new OLWmsCapabilities();
```
Comment (cmcnicholas) [inline src/polyfills/PolyfillWms.ts]:
```suggestion
  private static readonly caps = new OLWmsCapabilities();
```
